src/torchmetrics/metric.py

```diff
@@ -700,7 +702,9 @@ def state_dict(
             keep_vars: by default the :class:`~torch.Tensor`s returned in the state dict are detached from autograd.
                 If set to ``True``, detaching will not be performed.
         """
-        destination = super().state_dict(destination=destination, prefix=prefix, keep_vars=keep_vars)
+        destination: Dict = super().state_dict(
```

Comment:
```suggestion
        destination: Dict[str, Union[torch.Tensor, List, Any]] = super().state_dict(
```

I know that the Any typing basically makes the Union useless from a typing perspective, but it helps with auto-complete from IDEs